plugins/modules/ndo_fex_device.py

```diff
@@ -0,0 +1,267 @@
+
+- name: Query a FEX Device by name
+  cisco.mso.ndo_fex_device:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_test_template
+    name: ansible_fex_device
+    state: query
+  register: query_name
+
+- name: Query a FEX Device by UUID
+  cisco.mso.ndo_fex_device:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_test_template
+    uuid: '{{ create_fex_device.current.uuid }}'
+    state: query
+  register: query_uuid
+
+- name: Query all FEX Devices
+  cisco.mso.ndo_fex_device:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_test_template
+    state: query
+  register: query_all
+
+- name: Delete a FEX Device by name
+  cisco.mso.ndo_fex_device:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_test_template
+    name: ansible_test_fex_device
+    state: absent
+
+- name: Delete a FEX Device by UUID
+  cisco.mso.ndo_fex_device:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_test_template
+    uuid: '{{ create_fex_device.current.uuid }}'
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import (
+    MSOModule,
+    mso_argument_spec,
+)
+from ansible_collections.cisco.mso.plugins.module_utils.template import (
+    MSOTemplate,
+    KVPair,
+)
+from ansible_collections.cisco.mso.plugins.module_utils.utils import (
+    append_update_ops_data,
+)
+import copy
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        template=dict(type="str"),
+        template_id=dict(type="str"),
+        name=dict(type="str", aliases=["fex_device"]),
+        uuid=dict(type="str", aliases=["fex_device_uuid"]),
+        description=dict(type="str"),
+        fex_id=dict(type="int"),
+        nodes=dict(type="list", elements="str"),
+        interfaces=dict(type="list", elements="str"),
+        state=dict(type="str", default="query", choices=["absent", "query", "present"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["name", "uuid"], True],
+            ["state", "absent", ["name", "uuid"], True],
+            ["state", "present", ["fex_id", "nodes", "interfaces"]],
+        ],
+        required_one_of=[["template", "template_id"]],
+    )
+
+    mso = MSOModule(module)
+
+    template = module.params.get("template")
+    template_id = module.params.get("template_id")
+    name = module.params.get("name")
+    uuid = module.params.get("uuid")
+    description = module.params.get("description")
+    fex_id = module.params.get("fex_id")
+    nodes = module.params.get("nodes")
+    if nodes:
+        nodes = [str(node) for node in nodes]
```

Comment:
why use str() when nodes is a list that contains elements of type string?

Reply:
Good catch. Removed it!